api/proxy/store/generated_key/v2/eigenda.go

```diff
@@ -295,6 +307,48 @@ func (e Store) VerifyCert(ctx context.Context, versionedCert *certs.VersionedCer
 		return fmt.Errorf("eth-call to CertVerifier.checkDACert: %w", err)
 	}
 
+	// For cert versions that support offchain derivation versioning (v4+),
+	// we need to fetch the offchain derivation version from the contract,
+	// and then use that to get the relevant offchain derivation parameters
+	// (e.g. RBN recency window size) to perform additional checks.
+	//
+	// For cert versions that do not support offchain derivation versioning (v3 and below),
+	// we skip these additional checks.
+	//
+	// Note: offchain derivation versioning was introduced in cert version 4.
+	if certVersion >= coretypes.VersionFourCert {
+		timeoutCtx, cancel := context.WithTimeout(ctx, e.contractCallTimeout)
+		defer cancel()
+
+		offchainDerivationVersion, err := e.certVerifier.GetOffchainDerivationVersion(
+			timeoutCtx,
```

Comment:
wasn't bowen saying that we're going to add the derivation_version to the cert itself precisely so that we don't need to make call to the contract?

Reply:
Replaced call to certverifier with getting the value from the V4 cert in [d4b0403](https://github.com/Layr-Labs/eigenda/pull/2414/commits/d4b0403b0bcf737e7c3caa995c31651af390ee9f)